src/EventStore.Core/TransactionLog/Chunks/TFChunkManager.cs

```diff
@@ -80,7 +81,7 @@ public class TFChunkManager : IDisposable {
 				chunkNum = chunk.ChunkHeader.ChunkStartNumber - 1;
 			}
 
-			for (int chunkNum = lastChunkToCache; chunkNum < _chunksCount;) {
+			for (int chunkNum = lastChunkToCache; chunkNum < _chunks.Count;) {
 				var chunk = _chunks[chunkNum];
```

Comment:
we are outside of the lock here, is it possible that there are threading problems if we happen to write while we read? if not, say, because of the specific way we are using the list, it would be worth a comment where the list is declared explaining

Reply:
well spotted, thanks Tim. As discussed, i've simplified the PR for now: https://github.com/EventStore/EventStore/pull/3643#issuecomment-1340724427